pkg/azclient/armauth/keyvault_credential.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package armauth
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/policy"
+	"github.com/Azure/azure-sdk-for-go/sdk/keyvault/azsecrets"
+)
+
+type KeyVaultCredential struct {
+	secretClient *azsecrets.Client
+	secretPath   string
+
+	token *azcore.AccessToken
+}
+
+type KeyVaultCredentialSecret struct {
+	AccessToken string    `json:"access_token"`
+	ExpiresOn   time.Time `json:"expires_on"`
+}
+
+var secretURIRE = regexp.MustCompile("^(https://[a-zA-Z0-9-_]+\\.vault\\.azure\\.net)(/secrets/)([a-zA-Z0-9-_]+)/?$")
```

Comment:
this may not work in sovereign cloud.

Reply:
Good catch. Fixed and added test for `https://{vault-name}.vault.azure.cn`.